posthog/api/feature_flag.py

```diff
@@ -165,6 +166,9 @@ def update(self, instance: FeatureFlag, validated_data: Dict, *args: Any, **kwar
         instance = super().update(instance, validated_data)
         instance.update_cohorts()
 
+        if validated_data.get("deleted", False):
+            Experiment.objects.filter(feature_flag=instance).delete()
```

Comment:
@neilkakkar I'm removing hard deletes for (amongst other things) feature flags. I think this correctly replaces the cascade delete for experiments...

Reply:
Is everything moving towards soft deletes?

If so, does it make sense to do the same for experiments? Since now, a soft-delete on the feature flag will delete the experiment, but the undo won't bring the experiment back.

Reply:
Let's assume yes, but with low urgency :)

What makes sense for an experiment and its flag? Should their deletes be undo-able?

Reply:
hmm, okay.

In that case, I'd suggest not deleting an experiment at all when the FF is soft-deleted. Let it be manual instead of automatic, since people still might want the results they've had for an experiment, while deleting the FF to create a new one to use in a new experiment